addons/stock/tests/test_quant_inventory_mode.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import TransactionCase
+
+# Test only when user edit manually and directly a quant.
+class TestEditableQuant(TransactionCase):
+    def setUp(self):
+        super(TestEditableQuant, self).setUp()
+
+        self.Quant = self.env['stock.quant'].with_context(inventory_mode=True)
+
+        Product = self.env['product.product']
+        Location = self.env['stock.location']
+        self.product = Product.create({
+            'name': 'Product A',
+            'type': 'product',
+            'categ_id': self.env.ref('product.product_category_all').id,
+        })
+        self.product_tracked_sn = Product.create({
+            'name': 'Product tracked by SN',
+            'type': 'product',
+            'tracking': 'serial',
+            'categ_id': self.env.ref('product.product_category_all').id,
+        })
+        self.warehouse = Location.create({
+            'name': 'Warehouse',
+            'usage': 'internal',
+        })
+        self.stock = Location.create({
+            'name': 'Stock',
+            'usage': 'internal',
+            'location_id': self.warehouse.id,
+        })
+        self.room1 = Location.create({
+            'name': 'Room A',
+            'usage': 'internal',
+            'location_id': self.stock.id,
+        })
+        self.room2 = Location.create({
+            'name': 'Room B',
+            'usage': 'internal',
+            'location_id': self.stock.id,
+        })
+
+    def test_create_quant_1(self):
+        """ Create a new quant who don't exist yet.
+        """
+        # Checks we don't have any quant for this product.
+        quants = self.env['stock.quant'].search([('product_id', '=', self.product.id)])
+        self.assertEqual(len(quants), 0)
+        self.Quant.create({
+            'product_id': self.product.id,
+            'location_id': self.stock.id,
+            'inventory_quantity': 24
+        })
+        quants = self.env['stock.quant'].search([
+            ('product_id', '=', self.product.id),
+            ('quantity', '>', 0),
+        ])
```

Comment:
you have to check the create stock move
seach stock move product limit=1 assert done from inventory loss to loc